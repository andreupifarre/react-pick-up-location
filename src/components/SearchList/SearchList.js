import React from 'react';
import PropTypes from 'prop-types';
import './SearchList.scss';

const SearchList = ({ locations, noResults }) => (
  <div className="c-search-list">
    <ol className="c-search-list__list" role="listbox">
      {noResults && (
        <span className="c-search-list__nores" aria-live="assertive">
          No results found
        </span>
      )}
      {locations.map(loc => (
        <li
          key={loc.index}
          className="c-search-list__item"
          role="option"
          aria-selected="false"
          tabIndex="-1"
        >
          <div className="c-search-list__location-name">
            {loc.name} {loc.iata && `(${loc.iata})`}
          </div>
          <div className="c-search-list__support-text">
            {loc.city && loc.city + ', '}
            {!loc.city && loc.region && loc.region + ', '}
            {loc.country}
          </div>
        </li>
      ))}
    </ol>
    {locations.length > 0 && (
      <span className="c-search-list__reader-inst" aria-live="assertive">
        {locations.length} results are available.
      </span>
    )}
  </div>
);

SearchList.defaultProps = {
  locations: [],
  noResults: false,
};

SearchList.propTypes = {
  locations: PropTypes.array.isRequired,
  noResults: PropTypes.bool.isRequired,
};

export default SearchList;
