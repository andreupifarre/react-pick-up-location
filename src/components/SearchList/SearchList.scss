.c-search-list {
  &__list {
    position: absolute;
    z-index: 10;
    width: 100%;
    overflow: hidden;
    background: #fff;
    border-radius: 4px;
    -webkit-box-shadow: 0 1px 2px rgba(0, 0, 0, 0.2);
    box-shadow: 0 1px 2px rgba(0, 0, 0, 0.2);
    display: table;
    list-style: none;
    margin: 0;
    padding: 0;
  }

  &__nores {
    padding: 8px 16px;
    display: inline-block;
    font-style: italic;
  }

  &__item {
    cursor: pointer;
    padding: 8px 16px;
  }

  &__item:hover {
    background-color: #e7f4fe;
  }

  &__reader-inst {
    border: 0;
    clip: rect(0 0 0 0);
    height: 1px;
    margin: -1px;
    overflow: hidden;
    padding: 0;
    position: absolute;
    width: 1px;
  }
}
