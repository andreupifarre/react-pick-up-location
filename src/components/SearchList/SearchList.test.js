import React from 'react';
import { shallow } from 'enzyme';
import SearchList from './SearchList';

const render = props => shallow(<SearchList {...props} />);

describe('SearchList', () => {
  it('should contain an ol element with a role and a class .c-search-list__list', () => {
    const element = render().find('.c-search-list__list');
    expect(element.exists()).toBeTruthy();
    expect(element.name()).toEqual('ol');
    expect(element.prop('role')).toEqual('listbox');
  });

  it('should display a "No results found" message if no results where found', () => {
    const props = { locations: [], noResults: true };
    const element = render(props).find('.c-search-list__nores');
    expect(element.exists()).toBeTruthy();
  });

  it('should render list items correctly when locations array has items', () => {
    const props = {
      locations: [
        { index: 1, name: 'foo', iata: 'BCN' },
        { index: 2, name: 'bar', iata: 'MAD' },
      ],
      noResults: false,
    };

    const items = render(props).find('.c-search-list__item');
    expect(items.length).toEqual(props.locations.length);

    const propsFlat = props.locations.map(
      ({ name, iata }) => `${name} (${iata})`,
    );

    const newItems = items.map(item =>
      item
        .find('.c-search-list__location-name')
        .text()
        .trim(),
    );
    expect(newItems).toEqual(propsFlat);
  });

  it('should display a screen reader text indicating the results found', () => {
    const props = { locations: [{ index: 1 }], noResults: false };
    const element = render(props).find('.c-search-list__reader-inst');
    expect(element.exists()).toBeTruthy();
  });

  it('should render correctly', () => {
    expect(render().getElements()).toMatchSnapshot();
  });
});
