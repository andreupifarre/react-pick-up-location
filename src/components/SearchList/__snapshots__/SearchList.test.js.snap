// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchList should render correctly 1`] = `
Array [
  <div
    className="c-search-list"
  >
    <ol
      className="c-search-list__list"
      role="listbox"
    />
  </div>,
]
`;
